src/cli/commands/test/mod.rs

```diff
@@ -0,0 +1,169 @@
+#[cfg(test)]
+mod tests;
+
+use lazy_static::lazy_static;
+use regex::Regex;
+use std::error::Error;
+use std::fs::{File, self};
+use std::io::BufReader;
+use serde_json::Value;
+use std::path::Path;
+use std::{collections::HashMap, fmt::Display, io::Write, path::PathBuf, str::from_utf8};
+use cairo_rs::{
+	cairo_run::cairo_run,
+	hint_processor::{
+		builtin_hint_processor::{
+			builtin_hint_processor_definition::{BuiltinHintProcessor, HintFunc},
+			hint_utils::get_integer_from_var_name,
+		},
+		hint_processor_definition::HintReference,
+		proxies::{exec_scopes_proxy::ExecutionScopesProxy, vm_proxy::VMProxy},
+	},
+	serde::deserialize_program::ApTracking,
+	vm::{errors::{vm_errors::VirtualMachineError}},
+};
+use clap::{Args, ValueHint};
+use log::error;
+use serde::Serialize;
+
+use super::CommandExecution;
+use super::list::{path_is_valid_directory, ListArgs};
+
+
+use crate::cli::commands::execute::compile::compile;
+
+#[derive(Args, Debug)]
+pub struct TestArgs {
+	/// Path to a cairo program
+	#[clap(short, long, value_hint=ValueHint::FilePath, value_parser=path_is_valid_directory)]
+	pub program: PathBuf,
+}
+
+// fn is_cairo(path: &str) -> Result<PathBuf, String> {
+// 	let path = PathBuf::from(path);
+// 	if path.exists() && path.is_file() {
+// 		match path.extension() {
+// 			Some(ext) if ext == "cairo" => Ok(path),
+// 			_ => Err(format!("\"{}\" is not a cairo file", path.display())),
+// 		}
+// 	} else {
+// 		Err(format!("\"{}\" is not a valid file", path.display()))
+// 	}
+// }
+
+pub fn get_test_functions(compiled_path: &PathBuf) -> Result<Vec<String>, String> {
```

Comment:
Rather than a `String` we prefer to use `enum` in order to manage errors.
Preferably using this crate: https://docs.rs/thiserror/latest/thiserror/

Reply:
oh this is a hackish solution because the signature of the exec have this following form `fn exec(&self) -> Result<TestOutput, String>`, do I still need to change the type?